.github/workflows/run_tests.yml

```diff
@@ -1,62 +1,100 @@
+  DEBIAN_FRONTEND: 'noninteractive'
+  HDF5_USE_FILE_LOCKING: 'FALSE'
+
 jobs:
-  test:
+  ultra_matrix_test:
+    name: Matrix Test of Python ${{ matrix.python-version }} on ${{ matrix.os }}
 
-    runs-on: ubuntu-latest
+    # Matrix driven OS
+    runs-on: ${{ matrix.os }}
+
+    # Default shell for ALL subsequent steps.
+    defaults:
+      run:
+        shell: bash {0}
+
+    # Defining matrix for OS and Python
     strategy:
+
+      # Ensure exhaustive testing of all scenarios to gather maximum amount of information
+      fail-fast: false
+
       matrix:
-        python-version: [3.6, 3.7, 3.8]
+        os: [ "macos-latest", "windows-latest", "ubuntu-18.04", "ubuntu-20.04",  "macos-10.15" ]
+        python-version: [ 3.6, 3.7, 3.8 ]
         test-name:
           - integration-test
 
+    # Main steps for the test to be reproduced across OS x Python
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v1
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install flake8 pytest coverage coveralls pytest-cov
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-        pip install -e .
-    - name: Lint with flake8
-      run: |
-        # stop the build if there are Python syntax errors or undefined names
-        #flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with pytest
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        COVERALLS_FLAG_NAME: ${{ matrix.test-name }}
-        COVERALLS_PARALLEL: true
-      run: |
-         ivadomed_download_data -d data_testing -o data_testing
-         ivadomed_download_data -d data_functional_testing -o data_functional_testing
-         pytest . -v --cov ivadomed/ --cov-report term-missing
-         coveralls --service=github
+      # Step 0: Checkout code.
+      - uses: actions/checkout@v2
 
+      # Step 1: Setup python version
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      # Step 2: Install requirement dependencies.
+      - name: Install dependencies
+        uses: py-actions/py-dependency-install@v2
+
+      # Step 3: install additional pytest/dev related dependencies
+      - name: Install dev dependencies
+        run: |
+          pip install flake8 pytest coverage coveralls pytest-cov pytest-console-scripts pytest-ordering
+          pip install torch==1.5.0+cpu torchvision==0.6.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
```

Comment:
@dyt811 Consider using the previous step instead of hardcoding package versions.

Reply:
Good point!